certbot/certbot/main.py

```diff
@@ -1,8 +1,12 @@
 """Certbot main public entry point."""
+from typing import List
+from typing import Optional
+from typing import Union
+
 from certbot._internal import main as internal_main
 
 
-def main(cli_args=None):
+def main(cli_args: Optional[List[str]] = None) -> Optional[Union[str, int]]:
```

Comment:
Yes when `main` is called from an CLI entrypoint, it cannot be `None`. But in theory one could call this function directly in some python code, and then it could be `None`.